File: app/src/main/java/com/lucciola/haskellanywhereforandroid/controller/SymbolButtonController.java
package com.lucciola.haskellanywhereforandroid.controller;

import android.widget.Button;

import com.lucciola.haskellanywhereforandroid.View;
import com.lucciola.haskellanywhereforandroid.ProgramManager;

/**
 * Created by hiden on 2017/06/23.
 */

public class SymbolButtonController extends Controller {

    public SymbolButtonController(View arg0, ProgramManager arg2) {
        super(arg0, arg2);
    }

    @Override
    public void onClick(android.view.View v) {
        String symbol = (String) ((Button) v).getText();
        this.view.instertInputTextCurrentPositon(symbol);
    }

}

File: app/src/main/kotlin/com/lucciola/haskellanywhereforandroidkt/controller/SymbolButtonController.kt
package com.lucciola.haskellanywhereforandroidkt.controller

import android.widget.Button

import com.lucciola.haskellanywhereforandroidkt.View
import com.lucciola.haskellanywhereforandroidkt.ProgramManager

/**
 * Created by hiden on 2017/06/23.
 */

class SymbolButtonController(arg0: View, arg2: ProgramManager) : Controller(arg0, arg2) {

    override fun onClick(v: android.view.View) {
        val symbol = (v as Button).text as String
        this.view.instertInputTextCurrentPositon(symbol)
    }

}
